File: edupath-back/src/main/java/com/edupath/adapter/UADEAdapter.java
package com.edupath.adapter;

import com.edupath.interfaces.IAlumnoRepository;
import com.edupath.model.Alumno;
import com.edupath.model.RegistroAcademico;
import com.edupath.service.SistemaUadeAPI;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public class UADEAdapter implements IAlumnoRepository {

    private final SistemaUadeAPI cliente;

    public UADEAdapter(SistemaUadeAPI cliente) {
        this.cliente = cliente;
    }

    @Override
    public Alumno buscarPorLegajo(String legajo) {
        return cliente.getHistorialAlumno(legajo);
    }

    @Override
    public List<RegistroAcademico> obtenerRegistros(String legajo) {
        return cliente.getRegistrosAlumno(legajo);
    }
}

File: edupath-front/src/app/repositories/uade.adapter.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import { IAlumnoRepository } from './i-alumno.repository';
import { Alumno } from '../models/alumno.model';
import { RegistroAcademico } from '../models/registro-academico.model';

const API = 'http://localhost:8080/api/plan';

@Injectable({ providedIn: 'root' })
export class UadeAdapter implements IAlumnoRepository {

  constructor(private http: HttpClient) {}

  buscarPorLegajo(legajo: string): Observable<Alumno> {
    return this.http.get<Alumno>(`${API}/${legajo}/alumno`);
  }

  obtenerRegistros(legajo: string): Observable<RegistroAcademico[]> {
    return this.http.get<RegistroAcademico[]>(`${API}/${legajo}/registros`);
  }
}
